authz

**Describe the bug**
```
cannot build app:

	cannot build app:

	error while running command /tmp/tsc3926141537: : exit status 1

vue/src/store/generated/cosmos/cosmos-sdk/cosmos.staking.v1beta1/index.ts(6,10): error TS2305: Module '"./module/types/cosmos/staking/v1beta1/authz"' has no exported member 'Validators'.
```

**To Reproduce**
Steps to reproduce the behavior:
1. git clone https://github.com/notional-labs/dig
2. starport chain serve

**Please provide output of commands**
 - `starport version`
 starport version v0.17.3 linux/amd64 -build date: 2021-08-19T09:13:08Z
 - `go version`
 go version go1.17 linux/amd64
 - `uname -a`
`Linux archive 5.13.13-arch1-1 #1 SMP PREEMPT Thu, 26 Aug 2021 19:14:36 +0000 x86_64 GNU/Linux`

--- a/starport/pkg/protoanalysis/builder.go
+++ b/starport/pkg/protoanalysis/builder.go
@@ -86,7 +86,7 @@ func (b builder) buildMessages() (messages []Message) {
 		}
 	}
 
-	return
+	return messages
 }
 
 func (b builder) toServices(ps []*proto.Service) (services []Service) {

--- a/starport/pkg/protoanalysis/builder.go
+++ b/starport/pkg/protoanalysis/builder.go
@@ -1,6 +1,7 @@
 package protoanalysis
 
 import (
+	"fmt"
 	"regexp"
 	"strings"
 
@@ -56,8 +57,29 @@ func (b builder) buildMessages() (messages []Message) {
 				}
 			}
 
+			// some proto messages might be defined inside another proto messages.
+			// to represents these types, an underscore is used.
+			// e.g. if C message inside B, and B inside A: A_B_C.
+			var (
+				name   = message.Name
+				parent = message.Parent
+			)
+			for {
+				if parent == nil {
+					break
+				}
+
+				parentMessage, ok := parent.(*proto.Message)
+				if !ok {
+					break
+				}
+
+				name = fmt.Sprintf("%s_%s", parentMessage.Name, name)
+				parent = parentMessage.Parent
+			}
+
 			messages = append(messages, Message{
-				Name:               message.Name,
+				Name:               name,
 				Path:               f.path,
 				HighestFieldNumber: highestFieldNumber,
 			})

--- a/starport/pkg/protoanalysis/protoanalysis_test.go
+++ b/starport/pkg/protoanalysis/protoanalysis_test.go
@@ -7,6 +7,16 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestNestedMessages(t *testing.T) {
+	packages, err := Parse(context.Background(), nil, "testdata/nested_messages")
+	require.NoError(t, err)
+
+	pkg := packages[0]
+	require.Equal(t, "A", pkg.Messages[0].Name)
+	require.Equal(t, "A_B", pkg.Messages[1].Name)
+	require.Equal(t, "A_B_C", pkg.Messages[2].Name)
+}
+
 func TestLiquidity(t *testing.T) {
 	packages, err := Parse(context.Background(), nil, "testdata/liquidity")
 	require.NoError(t, err)

--- a/starport/pkg/protoanalysis/testdata/nested_messages/nested_messages.proto
+++ b/starport/pkg/protoanalysis/testdata/nested_messages/nested_messages.proto
@@ -1,0 +1,11 @@
+syntax = "proto3";
+
+package nested_messages;
+
+message A {
+    message B {
+        message C {
+            hello string = 1;
+        }
+    }
+}

